File: Games/GameOfGames/src/main/java/jokatu/game/games/gameofgames/views/GameOfGamesViewResolver.java
package jokatu.game.games.gameofgames.views;

import jokatu.game.games.gameofgames.game.GameOfGames;
import jokatu.game.player.StandardPlayer;
import jokatu.game.viewresolver.ViewResolver;
import org.jetbrains.annotations.NotNull;
import org.springframework.web.servlet.ModelAndView;

class GameOfGamesViewResolver extends ViewResolver<StandardPlayer, GameOfGames> {
	GameOfGamesViewResolver(GameOfGames game) {
		super(game);
	}

	@NotNull
	@Override
	protected ModelAndView getDefaultView() {
		return new ModelAndView("views/game_admin");
	}

	@NotNull
	@Override
	protected Class<StandardPlayer> getPlayerClass() {
		return StandardPlayer.class;
	}

	@NotNull
	@Override
	protected ModelAndView getViewFor(@NotNull StandardPlayer player) {
		return getDefaultView();
	}
}

File: Games/GameOfGames/src/main/kotlin/jokatu/game/games/gameofgames/views/GameOfGamesViewResolver.kt
package jokatu.game.games.gameofgames.views

import jokatu.game.games.gameofgames.game.GameOfGames
import jokatu.game.player.StandardPlayer
import jokatu.game.viewresolver.ViewResolver
import org.springframework.web.servlet.ModelAndView

internal class GameOfGamesViewResolver(game: GameOfGames) : ViewResolver<StandardPlayer, GameOfGames>(StandardPlayer::class, game) {

	override val defaultView: ModelAndView
		get() = ModelAndView("views/game_admin")

	override fun getViewFor(player: StandardPlayer): ModelAndView {
		return defaultView
	}
}
